docs/developer-docs/latest/plugins/graphql.md

```diff
@@ -341,33 +339,207 @@ The object describing the extension accepts the following parameters:
+<!-- TODO: remove this comment — the link won't work until merged with PR #450 -->
+
+Policies directly implemented in `resolversConfig` are functions that take a `context` object and the `strapi` instance as arguments.
+The `context` object gives access to:
+
+* the `parent`, `args`, `context` and `info` arguments of the GraphQL resolver,
+* Koa's [context](https://koajs.com/#context) with `context.http` and [state](https://koajs.com/#ctx-state) with `context.state`.
+
+::: details Example of a custom GraphQL policy applied to a resolver
+
+```js
+
+// path: ./src/index.js
+
+module.exports = {
+  register({ strapi }) {
+    const extensionService = strapi.plugin('graphql').service('extension');
+
+    extensionService.use({
+      resolversConfig: {
+        'Query.categories': {
+          policies: [
+            (context, { strapi }) => {
+              console.log('hello', context.parent)
+              /**
+               * If 'categories' have a parent, the function returns true,
+               * so the request won't be blocked by the policy.
+               */ 
+              return context.parent !== undefined;
+            }
+          ],
+          auth: false,
+        },
+      }
+    })
+  }
+}
+```
+
+:::
+
+#### Middlewares
+
+[Middlewares](/developer-docs/latest/development/backend-customization/middlewares.md) can be applied to a GraphQL resolver through the `resolversConfig.middlewares` key.
+
+The `resolversConfig.middlewares` key is an array accepting a list of middlewares, each item in this list being either a reference to an already registered policy or an implementation that is passed directly (see [middlewares configuration documentation](/developer-docs/latest/development/backend-customization/routes.md#middlewares)).
+<!-- TODO: remove this comment — the link won't work until merged with PR #450 -->
+
+Middlewares directly implemented in `resolversConfig` can take the GraphQL resolver's `parent`, `args`, `context` and `info` objects as arguments.
+
+:::tip
+Middlewares with GraphQL can even act on nested resolvers, which offer a more granular control than with REST.
+:::
+
+:::details Examples of custom GraphQL middlewares applied to a resolver
+
+```js
+
+// path: ./src/index.js
+
+module.exports = {
+  register({ strapi }) {
+    const extensionService = strapi.plugin('graphql').service('extension');
+
+    extensionService.use({
+      resolversConfig: {
+        'Query.categories': {
+          middlewares: [
+            /**
+             * Basic middleware example #1
+             * publish categories by changing publicationState values
+             */
+            async (next, parent, args, context, info ) => {
```

Comment:
Not sure this example is really relevant. Especially because we're not publishing any category here. We're just updating the `publicationState` arg to always look for published categories.
We could remove this logic and focus on the `console.time` & `console.timeEnd` which allows monitoring the speed of our queries.